cns/restserver/util.go

```diff
@@ -790,6 +790,24 @@ func (service *HTTPRestService) getPrimaryHostInterface(ctx context.Context) (*w
 	return service.state.primaryInterface, nil
 }
 
+// getSecondaryHostInterface returns the cached InterfaceInfo, if available, otherwise
+// queries the IMDS to get the secondary interface info and caches it in the server-state before returning the result.
+func (service *HTTPRestService) getSecondaryHostInterface(ctx context.Context, macAddress string) (*wireserver.InterfaceInfo, error) {
+	if service.state.secondaryInterface == nil {
+		res, err := service.wscli.GetInterfaces(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get interfaces from wireserver client")
+		}
+		secondary, err := wireserver.GetSecondaryInterfaceFromResult(res, macAddress)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get secondary interface from wireserver client")
+		}
+
+		service.state.secondaryInterface = secondary
```

Comment:
can the cached interfaceInfo ever be outdated?

Reply:
removing this field as we discussed it would change for each call